### DISCUSSION TITLE
500 error on payment page in chrome, works fine in firefox

### DISCUSSION THREAD

**Discussion by okimann** [duncanmcclean/simple-commerce#840] [Help]
I have a commerce site setup and I tested it in firefox and no problem. In Chrome if i use the dummy gateway also no problem. but if i use stripe, wit with a real credit card the server returns with a 500 error. On my local server if i'm using the sandbox stripe account it works fine. just in production with a live credit card do i get the 500 error. Can't seem to figure out what's going on here. any idea where to start? 

nginx error log is (removed ip address and website urls):
The error log is "Primary script is unknown"
2023/03/01 00:02:44 [error] 274356#274356: *771551 FastCGI sent in stderr: "Primary script unknown" while reading response header from upstream, client: xxx.xxx.xxx.xx, server: website.com, request: "GET //xmlrpc.php?rsd HTTP/1.1", upstream: "fastcgi://unix:/var/run/php/php8.1-fpm.sock:", host: "website.com"
2023/03/01 01:15:44 [error] 598146#598146: *1465 FastCGI sent in stderr: "Primary script unknown" while reading response header from upstream, client: xxx.xxx.x.xx, server: website.com, request: "POST /wp-cron.php?doing_wp_cron=1677633343.8520510196685791015625 HTTP/1.1", upstream: "fastcgi://unix:/var/run/php/php8.1-fpm.sock:", host: "website.com", referrer: "https://website.com/wp-cron.php?doing_wp_cron=1677633343.8520510196685791015625"
2023/03/01 07:17:29 [error] 598146#598146: *14668 FastCGI sent in stderr: "Primary script unknown" while reading response header from upstream, client: xxx.xxx.xxx.xx, server: website.com, request: "GET /wordpress/wp-admin/setup-config.php?step=1 HTTP/1.1", upstream: "fastcgi://unix:/var/run/php/php8.1-fpm.sock:", host: "website.com"

No idea why in the end there it's trying to get a wordpress folder either. No wordpress sites have ever been installed on this server.

**Comment by okimann** [comment]
Ok found that the error is with the SSL, looking into that now
585 SSL_do_handshake() failed (SSL: error:141CF06C:SSL routines:tls_parse_ctos_key_share:bad key share) while SSL handshaking

**Comment by duncanmcclean** [comment]
My guess is that the WordPress error isn't related to your issue - just something trying to figure out what exploits you have (pretty normal).

The error for that will likely be found in your site's log file (storage/logs). It's weird it's only happening on Chrome and not Firefox though.

**Comment by okimann** [comment]
OK there were two issues, one with the ssl cert which i fixed and was unrelated to simple commerce.

The second issue i found in the log files (thanks for the tip! i forgot about that one) and it may be a minor bug. With the simple commerce template, if you turn off the dummy gateway in config/simple-commerce and use only the stripe payment sandbox, checkout throws an error of "Gateway is a required field". However, the simple commerce starter kit checkout/gateways/_stripe.antlers.html file doesn't have a gateway field defined. Once i added it everything worked perfectly. Do you want me to file a bug report for that?

**Reply by duncanmcclean** [reply]
There should be a hidden gateway field in the payment.antlers.html file which should get filled by some JavaScript.

**Comment by duncanmcclean** [comment]
Are you still having issues here?

**Reply by okimann** [reply]
Nope, all the issues were solved. I can't remember what the ssl cert issue was but we got it fixed. I added the hidden gateway field as you recommended so that one was solved as well. The hidden gateway field wasn't in the starter kit, but I saw it afterward in your documentation. Thanks!
